os/kernel/log_dump/log_dump.c

```diff
@@ -193,21 +245,93 @@ int log_dump_save(char ch)
 			current_ptr = 0; /* reset the read pointer as head is modified */
 			node->arr[0] = ch;
 			sq_addlast((sq_entry_t *)node, &log_dump_chunks);
+			if (set_comp_head) { /* new node, so compressed block starts at 0 */
+				node->comp_head = 0;
+				set_comp_head = false;
+			} else {
+				node->comp_head = -1;
+				set_comp_head = false;
+			}
 			log_dump_mem_check(max_size);
 		}
 
 	} else { /* there is place in tail */
-
+		struct log_dump_chunk_s *log_dump_tail = (struct log_dump_chunk_s *)sq_tail(&log_dump_chunks);
 		log_dump_tail->arr[curbytes] = ch;
 		curbytes += 1;
+		if (set_comp_head) { /* new node, so compressed block starts at 0 */
+			if (log_dump_tail->comp_head < 0) { /* compressed node not present */
+				log_dump_tail->comp_head = curbytes-1;
+			}
+			set_comp_head = false;
+		}
 		if (curbytes % CONFIG_LOG_DUMP_MEMCHECK_SIZE == 0) {
-			log_dump_mem_check(max_size);
+ 			log_dump_mem_check(max_size);
+			return LOG_DUMP_OK_HEAPCHK;
 		}
 	}
 
 	return LOG_DUMP_OK;
 }
 
+int log_dump_save(char ch)
+{
+	if (is_started_to_save == false) {
+		return LOG_DUMP_OK;
+	}
+
+	size_t free_size = 0;
+	int ret = 0;
+	uncomp_array[uncomp_curbytes] = ch;
+	uncomp_curbytes++;
+	if (uncomp_curbytes == CONFIG_LOG_DUMP_CHUNK_SIZE) {
+		/* compress the block, add it to the nodes, reset the uncomp_array */
+		set_comp_head = true;
+		size_t writesize = CONFIG_LOG_DUMP_CHUNK_SIZE;
+		unsigned char *out_buf = (unsigned char *)kmm_malloc(CONFIG_LOG_DUMP_CHUNK_SIZE + LZMA_PROPS_SIZE);
+		if(out_buf == NULL) {
+			return LOG_DUMP_MEM_FAIL;
+		}
+
+		ret = LzmaCompress(&out_buf[LZMA_PROPS_SIZE], &writesize, uncomp_array, CONFIG_LOG_DUMP_CHUNK_SIZE, out_buf, &propsSize, 0, 1 << 13 , -1, -1, -1, -1, 1);
+		if (ret != SZ_OK) {
+			kmm_free(out_buf);
+			return -ret;
```

Comment:
In this case, it can return `SZ_ERROR_DATA`, `SZ_ERROR_MEM` and so on.
But these values are not matched with log dump defines.
So in this case, we cannot distinguish the error case.
Could you check this?